ee/server/service/teams.go

```diff
@@ -935,6 +935,17 @@ func (svc *Service) editTeamFromSpec(
 	}
 	team.Config.MDM.MacOSSetup.EnableEndUserAuthentication = spec.MDM.MacOSSetup.EnableEndUserAuthentication
 
+	if spec.MDM.WindowsSettings.CustomSettings.Set {
+		if !appCfg.MDM.WindowsEnabledAndConfigured && len(team.Config.MDM.WindowsSettings.CustomSettings.Value) != len(spec.MDM.WindowsSettings.CustomSettings.Value) {
+			// TODO: Address potential edge cases when teams that previously utilized MDM features
+			// are edited later edited when MDM disabled
```

Comment:
Yeah I think that if windows MDM is not configured and the custom settings are changed and the new ones are not empty, it should result in an error? (curious why the corresponding macOS validation for that scenario is not already there?)

So you'd have to check more than comparing the length, but maybe you wanted to do that in a future PR?

Reply:
I _think_ this is the case for macOS as well, right? that why I was thinking about this as a future thing. I'll double check

Reply:
It is the same for macOS, but this might be a good opportunity to fix it. Probably just a loop, right? if all elements match, you're good.

For teams:

https://github.com/fleetdm/fleet/blob/951a029b9add0995fff5a579594ebd37ade4f79f/ee/server/service/teams.go#L1028-L1040

And for app config:

https://github.com/fleetdm/fleet/blob/8dca52c06acbc6954eeacbb34ba423b14841567c/server/service/appconfig.go#L627-L630

Reply:
Ah ok that's why I didn't find it for teams, it's using a map. Yeah you can take advantage of https://pkg.go.dev/slices#Equal (but you need to `slices.Sort` first).

Reply:
took a stab at this in [930ce9a](https://github.com/fleetdm/fleet/pull/15145/commits/930ce9a1851f532d67a35978c348dc9e0ebc58e0) the team specs for this need more work.